logger_config.py

```
@@ -0,0 +1,57 @@
+            self.handleError(record)
+
+class ImmediateFlushRotatingFileHandler(logging.handlers.RotatingFileHandler):
+    def emit(self, record):
+        try:
+            super().emit(record)
+            self.flush()
+        except Exception as e:
+            self.handleError(record)
+
+if os.path.exists("settings.json"):
+    with open("settings.json", "r") as settings_file:
```

Ruff (0.8.0)

Unnecessary open mode parameters

Remove open mode parameters

(UP015)